Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/conflict.py

```diff
@@ -0,0 +1,84 @@
+# Copyright (C) 2020-2024 Apple Inc. All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions
+# are met:
+# 1.  Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+# 2.  Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#
+# THIS SOFTWARE IS PROVIDED BY APPLE INC. AND ITS CONTRIBUTORS "AS IS" AND
+# ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+# WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+# DISCLAIMED. IN NO EVENT SHALL APPLE INC. OR ITS CONTRIBUTORS BE LIABLE FOR
+# ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+# DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+# SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+# CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+# OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+import sys
+
+from .command import Command
+from webkitcorepy import run
+
+
+class Conflict(Command):
+    name = 'conflict'
+    help = "Given the representative Radar ID of a conflicting merge, checkout the branch and setup the integration remote if needed."
+
+    INTEGRATION_REMOTE_NAME = "integration"
+    INTEGRATION_REMOTE_URL = "git@github.com:apple/WebKit-integration.git"
```

Comment:
I think we should do something a bit more complex, and generalized, here.

`git-webkit` defines a set of canonical remotes, accessible from the `repository.source_remotes()`. I think we should assume that an integration remote will always be `<source-remote>/<project>-integration`. We should check each source remote, there won't be many, and this will allow us to support Integration remotes on any source remote for any project.